tests/test_help.py

```diff
@@ -16,65 +16,50 @@
 
 import logging
 import textwrap
-from unittest.mock import patch
+from unittest.mock import patch, call
 
 import pytest
 
 from charmcraft import helptexts, main
-from charmcraft.main import COMMAND_GROUPS, Dispatcher, ArgumentParsingError
-from charmcraft.commands.version import VersionCommand
+from charmcraft.main import Dispatcher, ArgumentParsingError, HelpCommand
 from tests.factory import create_command
 
 
-# -- verifications on different short help texts
-
-all_commands = list.__add__(*[commands for _, _, commands in COMMAND_GROUPS])
-
-
 @pytest.fixture
 def help_builder():
     """Provide a clean and fresh help_builder instance, ensuring the module also has it."""
     help_builder = helptexts.help_builder = main.help_builder = helptexts.HelpBuilder()
     return help_builder
 
 
-@pytest.mark.parametrize("command", all_commands)
-def test_aesthetic_help_msg(command):
-    """All the real commands help msg start with uppercase and doesn't end with a dot."""
-    msg = command.help_msg
-    assert msg[0].isupper() and msg[-1] != "."
-
+# -- bulding of the "usage" help
 
-@pytest.mark.parametrize("command", all_commands)
-def test_aesthetic_args_options_msg(command, config):
-    """All the real commands args help messages start with uppercase and dont' end with a dot."""
 
-    class FakeParser:
-        """A fake to get the arguments added."""
-
-        def add_mutually_exclusive_group(self, *args, **kwargs):
-            """Return self, as it is used to add arguments too."""
-            return self
-
-        def add_argument(self, *args, **kwargs):
-            """Verify that all commands have a correctly formatted help."""
-            help_msg = kwargs.get("help")
-            assert help_msg, "The help message must be present in each option"
-            assert help_msg[0].isupper() and help_msg[-1] != "."
+def test_get_usage_message_with_command(help_builder):
+    """Check the general "usage" text passing a command."""
+    help_builder.init("testapp", "general summary", [])
+    text = help_builder.get_usage_message("build", "bad parameter for the build")
```

Comment:
I know this is just a test, but just in case :-)

```suggestion
    text = help_builder.get_usage_message("build", "bad parameter for build")
```